crates/release_plz_core/src/git/backend.rs

```diff
@@ -642,30 +638,79 @@ impl GitClient {
         Ok(())
     }
 
-    async fn get_pr_info_and_categorize_labels(
+    /// Retrieves and categorizes labels for a PR, ensuring case-insensitive deduplication
```

Comment:
why do you do case-insensitive deduplication?

I just tested on gitea, and it's possible to create labels that are both lower case and upper case.

Why should release-plz forbid this?

![image](https://github.com/user-attachments/assets/c2a47eca-d0b2-451e-a9dd-5e95661216ea)

Reply:
I was looking at it from the perspective of minimizing duplication.
Assuming the PR already had the label "BUG" created and assigned to a PR.
Let's say I re-ran release-pr and passed in the pr_label - "Bug", "bug".  it won't create a new label because those words are the same.

Reply:
what do you think

Reply:
I think that's up to gitea/github to decide. We shouldn't limit what users can do.
Plus removing this check will simplify the codebase.
So I would allow users to create labels with different capitalization 👍

Reply:
Okay. @MarcoIeni I have made the necessary changes. And also updated the test. let me know if there is anything